### PR TITLE
Fix all tests

### PR DESCRIPTION
**Pull request by mohamed82008**
This PR fixes all the Turing tests except the PMMH and IPMCMC numerical tests. It relies on https://github.com/TuringLang/Bijectors.jl/pull/33. So after the Bijectors PR is merged and a release is made, the Project and Manifest files here should be updated then this PR should pass tests on Travis.

### CODE REVIEW

**Comment by yebai** [comment]
Thanks, @mohamed82008. Is this ready for a review?

**Comment by mohamed82008** [comment]
Yes this is ready for a first review. The WIP is to reflect that tests fail even though the PR is called "Fix all tests" :) This is because of the Bijectors PR linked. I won't have time to work on either until this weekend though. But the Bijectors PR is simple enough and probably just needs the Tracker benchmark to get it merged.

**Comment by mohamed82008** [comment]
I merged with master locally and updated AHMC and there seems to be new test failures. I will look into it. Tests also seem to be taking excessively long. I think we may have a speed regression somewhere.

**Comment by mohamed82008** [comment]
It could also be related to the long compilation time that Chris reported on Slack.

File: src/core/ad.jl [hunk 1]
```diff
@@ -433,3 +475,53 @@ MvNormal(m::TrackedVector{<:Real}, A::UniformScaling{<:Real}) = MvNormal(m, A.λ
...
+        Tracker.@grad function $F(
+            dist::Dirichlet, 
+            x::Tracker.TrackedArray, 
+            ::Type{Val{proj}}
+        ) where {proj}
+            x_data = Tracker.data(x)
+            T = eltype(x_data)
+            y = $F(dist, x_data, Val{proj})
+            return  y, Δ -> begin
+                out = (ForwardDiff.jacobian(x -> $F(dist, x, Val{proj}), x_data)::Matrix{T})' * Δ
```
Comment (yebai) [inline src/core/ad.jl]: Is this a duplicate fix for https://github.com/TuringLang/Bijectors.jl/pull/28?

**Comment by yebai** [comment]
@mohamed82008 It seems Travis test halts after `AHMC` is compiled, see https://travis-ci.org/TuringLang/Turing.jl/jobs/561920312#L679. There might be something else slow in the test.

**Comment by mohamed82008** [comment]
This PR requires a new release of Tracker and Zygote. Some tests in stdlib/distributions.jl are still taking a very long time, so I need to look into it. I suspect it has to do with Dirichlet.

**Comment by mohamed82008** [comment]
After https://github.com/FluxML/Tracker.jl/issues/43 and https://github.com/FluxML/Zygote.jl/issues/302 are resolved, this PR should pass almost all tests. Currently, one SGLD test is giving an error, and four numerical PMMH and IPMCMC tests are failing on my machine.

**Comment by yebai** [comment]
> After FluxML/Tracker.jl#43 and FluxML/Zygote.jl#302 are resolved, this PR should pass almost all tests. 

Perhaps @willtebbutt can help with making a new release of Tracker and Zygote.

**Comment by mohamed82008** [comment]
Actually, I think the Zygote release is not important, only the Tracker.

**Comment by mohamed82008** [comment]
So the tests are passing now but are taking some time. I can look into the time issue in another PR.

File: src/core/ad.jl (hunk 1 above)
Comment (yebai) [reply src/core/ad.jl]: @mohamed82008

Comment (mohamed82008) [reply src/core/ad.jl]: No we won't need `ForwardDiff`, we can replace it with the `link_jacobian` or `invlink_jacobian` functions.

Comment (yebai) [reply src/core/ad.jl]: Do you want to remove these functions from this PR, or keep them until TuringLang/Bijectors.jl#28 is merged?

Comment (mohamed82008) [reply src/core/ad.jl]: I would keep them for now as a stop gap. This fixes the problem but can be improved when the Bijectors PR is merged.

Comment (mohamed82008) [reply src/core/ad.jl]: We are depending on ForwardDiff anyways so nothing to lose.

**Pull request merged**